### ISSUE TITLE
feat: support passing directories to `deno check`

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/deno#20813]
I find myself often ending up in a situation where I want to check if all types in a project are valid. Instinctively, I usually type `deno check` and then I'm getting an error that I didn't pass arguments. In my mind I'm thinking "ah right, so I want to type check everything in the current directory". This leads me to type `deno check .`, but this throws another error because directories are not supported. But shoot, I don't have any file which imports every other file in the project. So I'm not sure how to type check the whole project with one command.

I found a workaround though: If you have an extensive test suite, there is a high chance that all files in the project are imported at some point. If that's the case then running `deno test` is a better mechanism to type check a project than `deno check`, which seems odd.

Feature request:
- Add support for passing directories to `deno check`
- If no arguments are provided, use the current directory `.` as the default argument.

**Comment by nktpro** [comment]
Seconded. This was on my wishlist as well, such that it's consistent with `deno test` and `deno lint`.

Currently I'd have to remember to resort to bash expansion i.e. `deno check **/*.ts`, but that doesn't work in a non-shell environment (such as making it a `deno task` in `deno.json`)

**Comment by lino-levan** [comment]
> Currently I'd have to remember to resort to bash expansion i.e. deno check **/*.ts, but that doesn't work in a non-shell environment (such as making it a deno task in deno.json)

This is actually not the case. Part of why I *love* `deno task` so much is because it is a cross platform shell. Glob expansion is supported.

**Comment by lino-levan** [comment]
I do support this proposal, and wanted to link https://github.com/denoland/deno/issues/20278 as well, since it seems to be in a similar vein. `deno check` should act like `deno fmt` by default, I don't see a reason for it not to.

**Comment by nktpro** [comment]
> > Currently I'd have to remember to resort to bash expansion i.e. deno check **/*.ts, but that doesn't work in a non-shell environment (such as making it a deno task in deno.json)
> 
> This is actually not the case. Part of why I _love_ `deno task` so much is because it is a cross platform shell. Glob expansion is supported.

It turned out you're absolutely right. That was added since Deno `1.34`. I must have missed the changelog when it came out.

**Comment by Solo-steven** [comment]
Hi I would like to take this issue as my first issue to deno, if no one is going to work on it, I hope it can be assigned to me, thanks ~

**Comment by ilovethensa** [comment]
Is this being worked on or should i(try to) implement this?

**Comment by zyx-db** [comment]
Hey, if this is an issue that isn't currently being worked on, I would love to try and work on it myself! Please let me know if this sounds fine.

**Comment by harrysolovay** [comment]
@lino-levan any thoughts on how to proceed given the contribution offers?

**Comment by lino-levan** [comment]
There's no need to ask before you work on something. Feel free to work on this if you want to!

**Comment by zyx-db** [comment]
That's great! I will take a stab at it :smile: !

**Comment by zyx-db** [comment]
Hey there, I was able to look through the source code and I believe I was able to find [where the arguments are being parsed](https://github.com/denoland/deno/commit/60f7034ad824af10b18cea7c19c518481bed1042). 

However, it seems like the functions/ methods being used to parse the arguments belong to different repositories, namely 'clap' and 'deno_core'. Would it be the best approach to raise these issues in those repositories? Or is there an alternative approach I may be able to look into? I would appreciate any feedback on how to proceed from here!
